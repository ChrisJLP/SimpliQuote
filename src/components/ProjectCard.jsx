// components/ProjectCard.jsx
const ProjectCard = ({ title, description }) => {
  return (
    <div
      className="mt-8 bg-[#EFEFEC] rounded-lg shadow-md p-6 
          w-[75vw] h-[60vh] max-w-[60vh] max-h-[60vh] 
          md:w-[50vh] md:h-[60vh]
          lg:w-[50vh] lg:h-[60vh]
          lg:mt-8
          overflow-auto"
    >
      <h2 className="text-2xl font-semibold mb-4 text-center text-shadow-sm">
        {title}
      </h2>
      <p className="text-gray-700 text-center">{description}</p>
    </div>
  );
};

export default ProjectCard;
