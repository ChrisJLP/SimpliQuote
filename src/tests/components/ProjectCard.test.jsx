import { render, screen } from "@testing-library/react";
import { describe, it, expect, beforeEach } from "vitest";
import ProjectCard from "../../components/ProjectCard";

describe("ProjectCard Component", () => {
  const mockProps = {
    title: "Test Project",
    description: "Test Description",
  };

  beforeEach(() => {
    // Clean up before each test
    document.body.innerHTML = "";
  });

  it("renders title and description correctly", () => {
    render(<ProjectCard {...mockProps} />);

    expect(
      screen.getByRole("heading", { name: mockProps.title })
    ).toBeInTheDocument();
    expect(screen.getByText(mockProps.description)).toBeInTheDocument();
  });

  it("matches snapshot for styling", () => {
    const { container } = render(<ProjectCard {...mockProps} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("handles long content gracefully", () => {
    const longProps = {
      title: "A".repeat(50),
      description: "B".repeat(200),
    };

    render(<ProjectCard {...longProps} />);
    const card = screen
      .getByRole("heading", { name: longProps.title })
      .closest("div");

    expect(card).toHaveClass("overflow-auto");
  });

  it("maintains responsive design classes", () => {
    render(<ProjectCard {...mockProps} />);
    const card = screen
      .getByRole("heading", { name: mockProps.title })
      .closest("div");

    // Test for responsive classes
    expect(card.className).toMatch(/md:w-\[50vh\]/);
    expect(card.className).toMatch(/lg:w-\[50vh\]/);
  });
});
