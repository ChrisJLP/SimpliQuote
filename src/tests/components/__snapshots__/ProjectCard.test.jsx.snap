// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProjectCard Component > matches snapshot for styling 1`] = `
<div
  class="mt-8 bg-[#EFEFEC] rounded-lg shadow-md p-6 
          w-[75vw] h-[60vh] max-w-[60vh] max-h-[60vh] 
          md:w-[50vh] md:h-[60vh]
          lg:w-[50vh] lg:h-[60vh]
          lg:mt-8
          overflow-auto"
>
  <h2
    class="text-2xl font-semibold mb-4 text-center text-shadow-sm"
  >
    Test Project
  </h2>
  <p
    class="text-gray-700 text-center"
  >
    Test Description
  </p>
</div>
`;
